securedrop_client/logic.py

```diff
@@ -372,10 +376,12 @@ def authenticated(self):
     def sync_api(self):
         """
         Grab data from the remote SecureDrop API in a non-blocking manner.
+
+        TODO: This should be removed once sync_api calls have been removed from all the different
```

Comment:
we could wait until all the sync_api calls are removed first and i could do this TODO and remove the metadata sync queue as part of this PR.  either way works for me.

Reply:
to reduce the number of open prs, i updated this pr (see update to the description) to include removal of the metadata sync queue. this helps us to avoid a state of having both the metadata sync queue and background syncs.